### PR TITLE
Document backoff options

### PR DESCRIPTION
**Pull request by dedemorton**
Adds backup options for LS and ES outputs (closes https://github.com/elastic/beats/issues/5419)

NOTE: I wasn't sure quite where to add the options in the yaml file.

Do we want to say anything about the backoff factor used here? Probably not since it's not configurable?

### CODE REVIEW

**Comment by dedemorton** [comment]
@urso I hope I got the details right this time. Right now we're ordering the options by how commonly they are used (with related options appearing next to each other). Not sure that's the best approach, but for now, I've added the options to an arbitrary location because I'm not sure how common it is for users to turn these knobs.

File: libbeat/_meta/config.reference.yml [hunk 1]
```diff
@@ -256,7 +256,17 @@ output.elasticsearch:
...
 
-  # Configure http request timeout before failing an request to Elasticsearch.
+  # The number of seconds to wait before attempting to connect to Elasticsearch
+  # after an initial connection error. The default is 1s. If additional errors
+  # occur, the wait time is increased exponentially for each try until
+  # backoff.max is met.
+  #backoff.init: 1s
+
+  # The maximum number of seconds to wait before attempting to connect to
+  # Elasticsearch after a connection error. The default is 60s.
```
Comment (urso) [inline libbeat/_meta/config.reference.yml]: Some more clarification:

`backoff.init` is applied on an network error. If connection was active, but connection gets lost or some other protocol/network error occurs, beats will disconnect and use `backoff.init`. After `backoff.init`, beats try to reconnect. If the reconnect fails, the backoff is increased exponentially up to `backoff.max`. After successful reconnect, the backoff timer is reset. On next network error, backoff will start with again with `backoff.init`.

Comment (urso) [reply libbeat/_meta/config.reference.yml]: @dedemorton SGTM

**Comment by dedemorton** [comment]
@urso This PR is ready for a final review.

**Comment by dedemorton** [comment]
@urso I had to rebase and then run `make update` again. I guess that's a good general best practice...that you should rebase a stale branch before running `make update` to pick up doc changes (even if GitHub isn't asking you to rebase).

**Pull request merged**